### DISCUSSION TITLE
How to calculate the Time, Equivalence queries and Membership queries required by the algorithm in 1 DFA

### DISCUSSION THREAD

**Discussion by VishalPatel43** [LearnLib/learnlib#108] [Q&A]
https://github.com/VishalPatel43/majorProject --> My Project (clone the repo and load into IntelliJ Idea)

First create Generate the Random DFA --> Uncomment the generate DFA function and Comment the Learning algorithm
After that comment the genrate DFA function and then uncomment the learning algorithm
If you want to run this src/main/java/com/experiment --> Main.java

My experiment NL* takes more time and RivestSchapire does not out perform the standard DFA
Can you review it?

Otherwise I have to thoroghly understand the Learnlib and Automatalib library

**Comment by mtf90** [comment]
Comparing the effective runtime between two learning algorithms is always tricky because it depends on various external factors. Maybe the NL* implementation is just not that efficient. You could spin this further and argue that a C-based implementation would pobably also be faster than a Java-based one, etc.

In practice, it is typically the performance of the SUL (i.e., the time it requires to answer the posed queries) that dominates the runtime. Therefore, algorithms are typically compared by the number of queries they pose (or the number of symbols within these queries). This allows one to focus on the algorithmic properties of the learners and ignore implementation details.

So while you could measure the runtime as a nice side-statistic, I wouldn't put too much emphasis on it.
